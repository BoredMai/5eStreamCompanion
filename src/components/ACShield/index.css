:local(.ac-shield) {
  background: no-repeat url('../../assets/images/ac_shield.png') 50% / 100%;
  image-rendering: pixelated;
  width: 8.4rem;
  height: 10.5rem;
}

:local(.ac-shield .score) {
  position: relative;
  top: 1.25rem;
}
