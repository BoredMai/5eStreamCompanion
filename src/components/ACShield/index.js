import cn from 'classnames';
import React, { Component } from 'react';
import PropTypes from 'prop-types';

import { acShield, score as acScore } from './index.css';
import { tag, score, whiteOutline } from '../../index.css';

export default class ACShield extends Component {
  render() {
    const { value } = this.props;

    return (
      <div className={acShield}>
        <div className={cn(tag, whiteOutline)}>AC</div>
        <div className={cn(acScore, score, whiteOutline)}>{value}</div>
      </div>
    );
  }

  static propTypes = {
    value: PropTypes.number.isRequired,
  };
}
