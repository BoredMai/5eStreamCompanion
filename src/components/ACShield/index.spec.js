import React from 'react';
import { shallow } from 'enzyme';
import ACShield from '.';

describe('<ACShield />', () => {
  let wrapper;
  const props = {
    value: 20,
  };

  beforeEach(() => {
    wrapper = shallow(<ACShield {...props} />);
  });

  it('renders properly', () => {
    expect(wrapper.getElement()).toMatchSnapshot();
  });
});
