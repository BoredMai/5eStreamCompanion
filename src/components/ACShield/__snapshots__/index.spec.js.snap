// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ACShield /> renders properly 1`] = `
<div
  className="acShield"
>
  <div
    className="tag whiteOutline"
  >
    AC
  </div>
  <div
    className="score score whiteOutline"
  >
    20
  </div>
</div>
`;
